packages/rest/src/lib/handlers/SequentialHandler.ts

```diff
@@ -80,38 +109,35 @@ export class SequentialHandler {
 		this.manager.emit(RESTEvents.Debug, `[REST ${this.id}] ${message}`);
 	}
 
+	/**
+	 * Delay all requests for the specified amount of time, handling global rate limits
+	 * @param time The amount of time to delay all requests for
+	 * @returns
+	 */
+	private async globalDelayFor(time: number): Promise<void> {
+		await sleep(time, undefined, { ref: false });
+		this.manager.globalDelay = null;
+	}
+
 	/**
 	 * Queues a request to be sent
 	 * @param routeId The generalized api route with literal ids for major parameters
 	 * @param url The url to do the request on
 	 * @param options All the information needed to make a request
+	 * @param body The object passed as body options, used to determine whether to sublimit
 	 */
-	public async queueRequest(routeId: RouteData, url: string, options: RequestInit): Promise<unknown> {
+	public async queueRequest(routeId: RouteData, url: string, options: RequestInit, body?: unknown): Promise<unknown> {
+		// TODO: Find better way to handle sublimits
+		const queue = isSublimitted(routeId.bucketRoute, body) ? this.#sublimitedQueue : this.#asyncQueue;
```

Comment:
I was thinking we could probably memoize and skip a lot of checks in the future if more sub-limits are added (sort of strategies?), so in most cases this will be a NOP.

We could also figure out something so a sub-limit queue isn't constructed if the route the `SequentialHandler` was created for doesn't support it.

Reply:
Explaining this in a new comment, but I think I took care of it